### ISSUE TITLE
matplotlib.backend_bases.Event: 'figure_leave_event' returns incorrect name

### ISSUE THREAD

**Issue by PeterJust** [matplotlib/matplotlib#18920]
In the event handling backend, the 'figure_leave_event' doesn't return the correct name. It returns 'motion_notify_event ' instead.
Code for demonstration:
```
import numpy as np
import matplotlib.pyplot as plt

def test_fun(event):
    print(event.name)

x = np.arange(1,10)
y = np.sin(x)

fig1 = plt.figure()
ax1 = plt.axes()
plt.plot(x,y)

fig1.canvas.mpl_connect('figure_leave_event', test_fun)

plt.show()
```
Cheers
Peter

**Comment by anntzer** [comment]
fixed by https://github.com/matplotlib/matplotlib/pull/16931 (in particular).

**Comment by ksunden** [comment]
This has been fixed, probably by #16931, as mentioned previously that it would fix it.
